### PR TITLE
Exposed log capture functionality to MinimalLib

### PR DESCRIPTION
**Pull request by ptosco**
- exposed to CFFI and JS MinimalLib functionality to capture RDKit logs with two modalities:
  1. `tee`: the log is captured in a buffer but also sent to the native output channel
  2. `capture`: the log is captured in a buffer without being sent to the native output channel
- removed duplicate `logging_needs_init` and `needs_init` atomic bool variables from `cffiwrapper.cpp` and consolidated them into a static `LoggingFlag d_loggingNeedsInit` class variable
- added relevant C and JS tests

### CODE REVIEW

File: Code/MinimalLib/cffi_test.c [hunk 1]
```diff
@@ -2164,6 +2164,51 @@ void test_partial_sanitization() {
...
+  char *mpkl;
+  char *log_buffer;
+  void *null_handle = NULL;
+  size_t mpkl_size;
+  void *log_handle;
+  typedef struct {
+    const char *type;
+    void *(*func)(const char *);
+  } capture_test;
+  capture_test tests[] = {{"tee", set_log_tee}, {"capture", set_log_capture}};
```
Comment (greglandrum) [inline Code/MinimalLib/cffi_test.c:2179]: Is this legal C?

Comment (ptosco) [reply Code/MinimalLib/cffi_test.c:2179]: Are you referring to the structure initialization? If so, yes, it is C99:
https://en.cppreference.com/w/c/language/struct_initialization

**Comment by greglandrum** [comment]
@ptosco : is the last "wip" commit here because this PR isn't ready to be reviewed? If so it's easiest to just mark it as a draft.

**Comment by ptosco** [comment]
@greglandrum sorry, I forgot that this PR was still open and I used this branch for rather unrelated commits. I’ll revert them shortly.

**Comment by ptosco** [comment]
@greglandrum Done, sorry for the confusion.

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**